src/piptool.py

```diff
@@ -137,22 +138,16 @@ def main():
 
     reqs = sorted(get_requirements(args.input), key=as_tuple)
     python_version = "%d%d" % (sys.version_info[0], sys.version_info[1])
-    whl_targets = []
+    whl_targets = {}
```

Comment:
please use OrderedDict

Reply:
added import with `try .. except ImportError: ...` to be forward-compatible with 3.10